exporter/metric/metric.go

```diff
@@ -66,7 +65,9 @@ type metricExporter struct {
 	// TODO: [ymotongpoo] this map should be goroutine safe with mutex.
 	mdCache map[key]*googlemetricpb.MetricDescriptor
 
-	client *monitoring.MetricClient
+	client monitoringpb.MetricServiceClient
+
+	cloudMock *cloudmock.CloudMock
```

Comment:
After reading the code, it seems that you don't really need this field, just client should be enough.